examples/browser/src/components.ts

```diff
@@ -22,12 +22,12 @@ export function add(lhs: PartialVec2, rhs: PartialVec2): Vec2 {
 
 /** Subtracts two 2D vectors. */
 export function sub(lhs: PartialVec2, rhs: PartialVec2): Vec2 {
-  return { x: (lhs.x ?? 0) - (rhs.x ?? 0), y: (lhs.y ?? 0) - (rhs.y ?? 0) };
+  return add(lhs, scale(rhs, -1));
 }
 
 /** Scales a 2D vector. */
-export function scale(lhs: PartialVec2, rhs: number): Vec2 {
-  return { x: (lhs.x ?? 0) * rhs, y: (lhs.y ?? 0) * rhs };
+export function scale({ x = 0, y = 0 }: PartialVec2, rhs: number): Vec2 {
```

Comment:
If we default these parameters, the vector no longer needs to be a `PartialVec2`, right?

```suggestion
export function scale({ x = 0, y = 0 }: Vec2, rhs: number): Vec2 {
```

Also it would be great if we could use this trick for `add` and `sub` too, for consistency.